File: src/MrHotel/ApiService/Rooms/Data/RoomAvailability.cs
namespace MrHotel.ApiService.Rooms.Data;

using MrHotel.Database.Entities.Reservations;
using MrHotel.Database.Entities.Rooms;

public record RoomAvailability(
    RoomInfo Room,
    RoomAvailabilityState State,
    ReservationInfo? ActiveReservation);

File: src/MrHotel/WebApp/src/app/services/room-manager/data/room-availability.ts
import { ReservationInfo } from "@services/reservation-manager/data";
import { RoomAvailabilityState, RoomInfo } from ".";
import { Nullable } from "@customTypes/nullable";

export type RoomAvailability = {
  room: RoomInfo;
  state: RoomAvailabilityState;
  activeReservation: Nullable<ReservationInfo>;
};
